Cython/Utility/ModuleSetupCode.c

```diff
@@ -441,6 +441,12 @@ class __Pyx_FakeReference {
   #define PyObject_Realloc(p)  PyMem_Realloc(p)
 #endif
 
+#if CYTHON_COMPILING_IN_CPYTHON && PY_VERSION_HEX < 0x03040000
```

Comment:
No need to restrict this to CPython. `PyMem_Malloc()` & friends should generally be available. I would add a `&& !defined(PyMem_Malloc)`, though, just in case. PyPy uses C-defines for all of the C-API, for example.

Also, more accurately, these functions were first added in `0x030400A1`, as you can see from the tags here: https://github.com/python/cpython/commit/0507bf56f0b1064a179464eb3a9d524fd6dad815

Reply:
Do the PyPy ones work without the GIL? Don't really know the PyPy C API so had left them out.